### PR TITLE
WIP: Notification System

### PR DESCRIPTION
**Pull request by aquibbaig**
This commit is WIP. Currently aimed at fixing #373 #374 #379

### CODE REVIEW

**Comment by aquibbaig** [comment]
Screenshot 2019-06-13 at 3 30 16 PM

Will be coming up with a design for the UI today at the earliest, but this is what is being achievable rn, with all the configuration as discussed @imphil @agathver

**Comment by aquibbaig** [comment]
Screenshot 2019-06-14 at 10 18 54 PM

## Status

- Added Mark as Read and Remove Notifications with their specific handlers. Note that these features do not depend on other parameters than the notification itself such as user, project, etc. So to make it independent, it is wiser to create another Controller with specific routes to handle it.
- Also, the color and styles are subject to change and to be moved to an independent file after #353

File: site/.gitignore [hunk 1]
```diff
@@ -22,3 +22,4 @@
 /node_modules/*
 /web/build/*
 /yarn-error.log
+/assets/js/frontend-config.json
```
Comment (imphil) [inline site/.gitignore]: Please do a separate PR for changes like this, they should be merged independent of this PR, which is more experimental.

**Comment by imphil** [comment]
To clarify what I meant with "That forwarding should happen through RabbitMQ, not by calling two handlers from a single consumer." please have a look at this picture:

image

That's the same setup we have for our repo crawler currently.

**Comment by aquibbaig** [comment]
@imphil Found some time out and made the required architectural changes. Probably have left out some PhPDocs comments. will look it up when they allow me to

**Comment by aquibbaig** [comment]
This Pull Request will be open to discussions only

**Comment by imphil** [comment]
This has now been merged through individual PRs, closing.
